pvanalytics/tests/features/test_daytime.py

```diff
@@ -182,3 +246,140 @@ def test_daytime_variable(clearsky_january):
+                              "right-aligned data, or 'C' for "
+                              "center-aligned data.")):
+        daytime.get_sunrise(daytime_mask_left_aligned,
+                            data_alignment='M')
+
+
+def test_sunset_alignment_error(daytime_mask_left_aligned):
+    with pytest.raises(ValueError,
+                       match=("No valid data alignment given. Please pass 'L'"
+                              " for left-aligned data, 'R' for "
+                              "right-aligned data, or 'C' for "
+                              "center-aligned data.")):
+        daytime.get_sunset(daytime_mask_left_aligned,
+                           data_alignment='M')
+
+
+def test_consistent_modeled_midday_series(daytime_mask_right_aligned,
+                                          daytime_mask_left_aligned,
+                                          daytime_mask_center_aligned,
+                                          modeled_midday_series):
+    # Get sunrise and sunset times for each time series (left, right, center),
+    # calculate the midday point, compared to modeled midday,
+    # and compare that it's consistent across all three series
+    # Right-aligned data
+    right_sunset = daytime.get_sunset(daytime_mask_right_aligned,
+                                      data_alignment='R')
+    right_sunrise = daytime.get_sunrise(daytime_mask_right_aligned,
+                                        data_alignment='R')
+    midday_series_right = right_sunrise + ((right_sunset - right_sunrise)/2)
+    midday_series_right.index = midday_series_right.index.date
+    midday_diff_right = (modeled_midday_series -
+                         midday_series_right.drop_duplicates())
+    # Left-aligned data
+    left_sunset = daytime.get_sunset(daytime_mask_left_aligned,
+                                     data_alignment='L')
+    left_sunrise = daytime.get_sunrise(daytime_mask_left_aligned,
+                                       data_alignment='L')
+    midday_series_left = left_sunrise + ((left_sunset - left_sunrise)/2)
+    midday_series_left.index = midday_series_left.index.date
+    midday_diff_left = (modeled_midday_series -
+                        midday_series_left.drop_duplicates())
+    # Center-aligned data
+    center_sunset = daytime.get_sunset(daytime_mask_center_aligned,
+                                       data_alignment='C')
+    center_sunrise = daytime.get_sunrise(daytime_mask_center_aligned,
+                                         data_alignment='C')
+    midday_series_center = center_sunrise + ((center_sunset -
+                                              center_sunrise)/2)
+    midday_series_center.index = midday_series_center.index.date
+    midday_diff_center = (modeled_midday_series -
+                          midday_series_center.drop_duplicates())
+    assert (midday_diff_right.equals(midday_diff_left) &
+            midday_diff_center.equals(midday_diff_right))
```

Comment:
In addition to verifying consistency across data labeling conventions, should we also check that the differences are small (less than some reasonable expected error)?

Reply:
Good idea. I added an asset statement to do this in https://github.com/pvlib/pvanalytics/pull/187/commits/e89e19c4109c42323f5426d272110ac08dd26bac. I use a 10 minute threshold as this was generally considered just noise when Bennet and I benchmarked in the time shift paper.